lib/barnes/instruments/puma_backlog.rb

```diff
@@ -26,22 +26,28 @@ def json_stats
         return {}
       end
 
-      # For single worker process use backlog directly
+      # For single worker process use value directly
       # for multiple workers sum backlog.
       #
       # https://github.com/puma/puma/pull/1532
-      def instrument!(state, counters, gauges)
-        stats = self.json_stats
-        return if stats.empty?
+      def stat_value_from_key(key)
+        stats = json_stats
+
+        value = stats[key]
+        return value if value
+
+        value = stats["worker_status"].map do |worker_status|
+          worker_status["last_status"][key] || 0
+        end.reduce(0, :+)
 
-        backlog = stats["backlog"]
-        if backlog.nil?
-          backlog = stats["worker_status"].map do |worker_status|
-            worker_status["last_status"]["backlog"]
-          end.reduce(0, :+)
-        end
+        return value
+      end
+
+      def instrument!(state, counters, gauges)
+        return if json_stats.empty?
 
```

Comment:
I'm not a ruby perf expert, but would think that `stat_value_from_key` could be rewritten to avoid multiple map / reduce pairs. Not sure how much it matters though, given the number of puma metrics is probably small.